sdk/metric/benchmark_test.go

```diff
@@ -313,43 +312,6 @@ func BenchmarkFloat64ExactAdd(b *testing.B) {
 	benchmarkFloat64HistogramAdd(b, "float64.exact")
 }
 
-// BatchRecord
-
-func benchmarkBatchRecord8Labels(b *testing.B, numInst int) {
-	const numLabels = 8
-	ctx := context.Background()
-	fix := newFixture(b)
-	labs := makeLabels(numLabels)
-	var meas []sdkapi.Measurement
-
-	for i := 0; i < numInst; i++ {
-		inst := fix.meterMust().NewInt64Counter(fmt.Sprintf("int64.%d.sum", i))
-		meas = append(meas, inst.Measurement(1))
-	}
-
-	b.ResetTimer()
-
-	for i := 0; i < b.N; i++ {
-		fix.accumulator.RecordBatch(ctx, labs, meas...)
-	}
-}
```

Comment:
It looks like RecordBatch is a form of optimization that was taken.  The alternatives seem to be calling Record N times for different synchronous instruments, and Observing() N Async instruments ina BatchObserver.

Could we please get a replacement benchmark to show the performance difference using these cases?

Reply:
As long as it is a major part of the API: new method/function on Meter + new method/function on all the instruments this needs to be in the specification in order for this to be compliant, so not sure why I need to do this.

Reply:
Also, I will send a separate PR for the BatchObserver.

Reply:
I understand the reason why we might not want to have to support this specific API, but there is a use case here.  Specifically "As a User I would like to record a number of metrics at once.", I just want to make sure that the performance of this use case hasn't degraded.

Reply:
Beyond simply "record a bunch of metrics at once", it's also "record a bunch of metrics that come from an expensive operation without performing that operation once per metric".

I think that regardless what is in the spec we have agreed that SIGs can choose to have more API surface than required and this may be a situation where we choose to do so.  The decision to add this capability was not taken lightly and I don't think the decision to remove it should be either.  I'd like to have this discussed at the Go SIG and definitely want input from @jmacd, who helped define the initial need for this capability and its implementation.